File: 347-top-k-frequent-elements/347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        count = {} # num, count
        freq = [[] for i in range(len(nums) + 1)] # array of length + 1 arrays 
        
        for num in nums:
            count[num] = 1 + count.get(num, 0)
            
        for key, value in count.items():
            freq[value].append(key)
        
        result = []
        
        for i in range(len(freq) -1 , 0, -1): #range(start, stop, increment) way to iterate by index
            for x in freq[i]:
                result.append(x)
                if len(result) == k:
                    return result

File: 347-top-k-frequent-elements/347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    const hash = {};
    let sortedArray = Array.from({length: nums.length + 1}, () => []);
    
    for (const num of nums) {
        if (!hash[num]) {
            hash[num] = 0;
        }
        hash[num]++;
    }
    
    for (const key in hash) {
        sortedArray[hash[key]].push(key);
    }
    console.log(sortedArray);
    
    const res = [];
    
    for (var i = sortedArray.length - 1; i > 0; i--) {
        for (const c of sortedArray[i]) {
            res.push(c);
            if (res.length === k) {
                return res;
            }
        }
    }
    return [];
};